plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -3251,7 +3262,12 @@ def self.rpc_find_all_available(opts=nil)
         end
 
         # Remove the least preferred servers from the list, ensuring there is at least one server remaining
-        server_infos.delete_if { |server_info| (server_infos.length > 1) && least_preferred_servers.include?(server_info.name) } if least_preferred_servers.present?
+        # Ignore least preferred servers if all servers are considered least preferred
+        if least_preferred_servers.present?
+          unless ignore_least_preferred_servers
```

Comment:
Do this in one line as 

if least_preferred_servers.present? && !ignore_least_preferred_servers